doc/adrs/168_add_support_for_forking_table_description_from_representation.adoc

```diff
@@ -0,0 +1,48 @@
+= ADR-166 Add support for forking table description from representation
+
+== Context
+
+On a table representation, we want to be able to modify the specification used as description of this table and then reload the table with this new specification.
+
+== Decision
+
+We will allow users to open a new studio from an opened table representation. This new studio will be a fork of the one used to instantiate the table. After modifying this new studio, the user will be able to re open the previous table representation that will use the updated studio as description.
+
+=== Back-end
+
+We will implement a new mutation `createForkedStudio` that will fork the description used by the table.
+`createForkedStudio(input: CreateForkedStudioInput!): CreateForkedStudioPayload!`
+```
+input CreateForkedStudioInput {
+  id: ID!
+  editingContextId: String!
+  representationId: String!
+}
+```
+
+The data fetcher will then call `this.createForkedStudioService.create(editingContextId, representationId)`
+
+This method will :
+
+* Dispose the current editing context
+* Create a new Studio that will be a fork of the description used by the representation
+* Update the representations in the database with the new descriptionId
+* Return the url of the newly created studio (so the front-end can redirect to this url)
+
+=== Front-end
+
+We will add an extension point to contribute to the settings menu item.
+One contribution will add a menu item in order to send the createForkedStudio mutation.
+
+== Advantages
+* This solution does not require deep modification on how we reload representation.
+
+== Disadvantages
+* User Experience: The user that wants to make a change will need to reopen his representation after the changes.
+* User Experience: Saved changes in the database from the previous representation will be irrelevant and deleted.
```

Comment:
Why? I don't see why anything should be irrelevant and deleted

Reply:
I'm not sure what data we're going to store for a table and wanted to avoid possible issues.
I thought about the case where if we store the current width of a column and that column does not exist anymore after some changes on the forked studio.